### ISSUE TITLE
Client ID not randomly generated 

### ISSUE THREAD

**Issue by JosuGZ** [rmqtt/rmqtt#27]
I think that when connecting with a client id of length 0, the server is not generating a proper id for the client, causing clients to kick each other from the network.

AFAIK the server must either allow a client id of length 0 an generate a random ID, or reject the client. I'm not sure if MQTT5 has changed this.

From the spec:

> A Server MAY allow a Client to supply a ClientId that has a length of zero bytes, however if it does so the Server MUST treat this as a special case and assign a unique ClientId to that Client. It MUST then process the CONNECT packet as if the Client had provided that unique ClientId [MQTT-3.1.3-6].

**Comment by try-box** [comment]
Yes, zero byte client identifiers are currently not supported. Are you sure you have a business that requires a zero byte client identifier?

**Comment by rmqtt** [comment]
Yes, zero byte client identifiers are currently not supported. Are you sure you have a business that requires a zero byte client identifier?

**Comment by JosuGZ** [comment]
In that case I think it should reject the connection:

> however if it does so the Server MUST treat this as a special case and assign a unique ClientId

I don't need this feature. I just found that one client was pushing the other out and had to figure out why.

I might take a look at the code :)

**Comment by rmqtt** [comment]
RMQTT 0.2.14 and later versions already support client identifiers with a length of 0, clean_session must be true
